src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAmendApplicationType.java

```diff
@@ -25,45 +40,85 @@
             .description("Amend application type")
             .grant(CREATE_READ_UPDATE, SUPER_USER)
             .grantHistoryOnly(LEGAL_ADVISOR, SOLICITOR, CASE_WORKER)
-            .aboutToSubmitCallback(this::aboutToSubmit);
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted))
+            .page("Amend application type")
+            .label("prayerChangeLabel","Completing this event will change the prayer on the case. "
+                + "Please make sure that you have confirmed this with the Applicant before proceeding.");
     }
 
     public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
                                                                        final CaseDetails<CaseData, State> beforeDetails) {
 
         log.info("Caseworker Amend Application Type about to submit callback invoked for case id: {}", details.getId());
         CaseData caseData = details.getData();
+        ApplicantPrayer app1Prayer = caseData.getApplicant1().getApplicantPrayer();
+        ApplicantPrayer app2Prayer = caseData.getApplicant2().getApplicantPrayer();
 
         if (isNull(caseData.getDivorceOrDissolution())) {
             return AboutToStartOrSubmitResponse.<CaseData, State>builder()
                 .data(caseData)
                 .errors(Collections.singletonList("divorceOrDissolution is null, cannot continue submitting event"))
                 .build();
-        } else if (caseData.isJudicialSeparationCase()) {
+        }
+
+        if (caseData.isJudicialSeparationCase()) {
             if (caseData.isDivorce()) {
                 caseData.setSupplementaryCaseType(SEPARATION);
+                if (JUDICIAL_SEPARATION_SET.equals(app1Prayer.getPrayerJudicialSeparation())) {
+                    app1Prayer.setPrayerSeparation(SEPARATION_SET);
+                }
+                if (JUDICIAL_SEPARATION_SET.equals(app2Prayer.getPrayerJudicialSeparation())) {
+                    app2Prayer.setPrayerSeparation(SEPARATION_SET);
+                }
             } else {
                 caseData.setSupplementaryCaseType(JUDICIAL_SEPARATION);
+                if (SEPARATION_SET.equals(app1Prayer.getPrayerSeparation())) {
+                    app1Prayer.setPrayerJudicialSeparation(JUDICIAL_SEPARATION_SET);
+                }
+                if (SEPARATION_SET.equals(app2Prayer.getPrayerSeparation())) {
+                    app2Prayer.setPrayerJudicialSeparation(JUDICIAL_SEPARATION_SET);
+                }
             }
         } else {
             if (caseData.isDivorce()) {
                 caseData.setDivorceOrDissolution(DISSOLUTION);
+                if (DISSOLVE_DIVORCE_SET.equals(app1Prayer.getPrayerDissolveDivorce())) {
+                    app1Prayer.setPrayerEndCivilPartnership(END_CIVIL_PARTNERSHIP_SET);
+                }
+                if (DISSOLVE_DIVORCE_SET.equals(app2Prayer.getPrayerDissolveDivorce())) {
```

Comment:
For these conditions, would the logic work if we check !NullOrEmpty rather than comparing against the sets defined as constants?:
```suggestion
                if (!NullOrEmpty(app2Prayer.getPrayerDissolveDivorce())) {
```